main.py

```diff
@@ -1152,3 +1152,45 @@ def handle_link_maintenance_end(self, event):
             link.endpoint_a.enable()
             link.endpoint_b.enable()
             self.notify_link_status_change(link, reason='maintenance')
+
+    @listen_to('topology.interruption.start')
+    def on_interruption_start(self, event: KytosEvent):
+        """Deals with the start of service interruption."""
+        with self._links_lock:
+            self.handle_interruption_start(event)
+
+    def handle_interruption_start(self, event: KytosEvent):
+        """Deals with the start of service interruption."""
+        interrupt_type = event.content['type']
+        # switches = event.content.get('switches', [])
+        # interfaces = event.content.get('interfaces', [])
+        links = event.content.get('links', [])
+        # for switch_id in switches:
+        #     pass
+        # for interface_id in interfaces:
+        #     pass
+        for link_id in links:
+            link = self.links[link_id]
+            self.notify_link_status_change(link, interrupt_type)
+        self.notify_topology_update()
+
+    @listen_to('topology.interruption.end')
+    def on_interruption_end(self, event: KytosEvent):
+        """Deals with the end of service interruption."""
+        with self._links_lock:
+            self.handle_interruption_end(event)
+
+    def handle_interruption_end(self, event: KytosEvent):
+        """Deals with the end of service interruption."""
+        interrupt_type = event.content['type']
+        # switches = event.content.get('switches', [])
+        # interfaces = event.content.get('interfaces', [])
+        links = event.content.get('links', [])
+        # for switch_id in switches:
+        #     pass
+        # for interface_id in interfaces:
+        #     pass
+        for link_id in links:
+            link = self.links[link_id]
+            self.notify_link_status_change(link, interrupt_type)
```

Comment:
@Ktmi, could you add a conditional safe guard checking if `link.status == EntityStatus.UP and not link.status_reason` before calling `self.notify_link_status_change(link, interrupt_type)`? Same [approach as we've been doing so far](https://github.com/kytos-ng/topology/blob/master/main.py#L595-L596), it turns out that `notify_link_status_change` without a safe guard checking for the status upfront it could send an unnecessary `kytos/topology.link_up` event which could have network converge implications on `mef_eline` (although it's going to try to discard redundant events) . We've been trying to maintain this consistently on topology to avoid sending redundant `kytos/topology.*` to simplify for subscribers and also to facilitate for us to follow the events when potentially debugging.